probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -326,7 +417,7 @@ impl<'interface> ArmCommunicationInterface {
         let mut ctrl_reg = Ctrl::default();
         ctrl_reg.set_csyspwrupreq(true);
         ctrl_reg.set_cdbgpwrupreq(true);
-        ctrl_reg.set_orun_detect(self.use_overrun_detect);
```

Comment:
I think we need overun detect, but it's not part of the default sequence. Do we change the default sequence or do we do it differently? If we change the default sequence, we need to do this for every sequence that we have :/

Also, further down below @line333 we check if powerup worked. I think we should do that too, but we can do it outside I think.

ore importantly is @line320 tho :/ here we set a clean state for the cached DP bank select and we also ensure it's actually 0 selected. I experienced that the debug probe can get out of sync with probe-rs occasionally, so I think this is required! Do we alter the default sequence or how do we handle this?

@Tiwalun @Dirbaio

Reply:
The `init_dp` function seems to be unused right now.

 I think setting the `orun_detect` should be done separately from the sequences, because the sequences can be changed, and we always want to use overrun detect for the J-Link. Error handling for SWD is a lot more difficult otherwise.

Regarding powerup, the original sequence(<https://arm-software.github.io/CMSIS_5/Pack/html/debug_description.html#debugPortStart>) contains a check for powerup as well.

Reply:
> The `init_dp` function seems to be unused right now.

Yes. I left it in there until this issue is resolved so we can easily use it for a side-by-side comparison :) It is replaced by the sequences init function.

> I think setting the `orun_detect` should be done separately from the sequences, because the sequences can be changed, and we always want to use overrun detect for the J-Link. Error handling for SWD is a lot more difficult otherwise.

Ok we can do this no prolem I think.

> Regarding powerup, the original sequence(https://arm-software.github.io/CMSIS_5/Pack/html/debug_description.html#debugPortStart) contains a check for powerup as well.

This only checks if it was powered down beforehand. It does not check if it's properly powered up afterwards :) Positive-assumption errohandling works ofc, but it's harder to do diagnostics on the error that follows on the next command because it's not powered up :)

Reply:
There is a check I think?

```
    <!-- Wait for Power-Up Request to be acknowledged -->
    <control while="(ReadDP(DP_CTRL_STAT) &amp; 0xA0000000) != 0xA0000000" timeout="1000000"/>
```

Do you mean something else?

Reply:
Ohwow, I missed that while completely ... mea culpa! I was just looking at the `powered_down` variable.

Reply:
Aand done :)